### DISCUSSION TITLE
How to add hint to dual variables?

### DISCUSSION THREAD

**Discussion by CenturionMKIII** [google/or-tools#4024] [Linear Solver questions]
Hi!

Is there any method to give an initial point for dual variables (just like add_hint for primal variables).
I tried to use add_hint but with constraints; it did not report any immediate error, but apparently it did not work as intended ('duplicated hint' when solving the model).

**Comment by rma350** [comment]
This operation is supported for MathOpt (see ModelSolveParameters) but not MPSolver/MPModelRequest. Note that simplex solvers prefer a basis to a hint and hints are to my knowledge not very useful for IPM solvers. PDLP, and I believe SCS and OSQP can all use hints well. Gurobi does try to consume a hint (by finding a near by basis). I believe this is also possible with GLOP but not exposed through any API (using the PDLP + GLOP crossover code).
